t/node/upstream-status-5xx.t

```diff
@@ -0,0 +1,359 @@
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "uri": "/hello",
+                        "upstream_id": "1"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 10: set upstream(id: 1, retries = 2), has available upstream
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/upstreams/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "nodes": {
+                        "127.0.0.3:1": 1,
+                        "127.0.0.2:1": 1,
+                        "127.0.0.1:1980": 1
+                    },
+                    "retries": 2,
+                    "type": "roundrobin"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 11: hit routes, status code is 200
+--- request
+GET /hello
+--- grep_error_log eval
+qr/X-Apisix-Upstream-Status:/
+--- grep_error_log_out
+
+
+
+=== TEST 12: set upstream(id: 1, retries = 2), all upstream nodes are unavailable
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/upstreams/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "nodes": {
+                        "127.0.0.3:1": 1,
+                        "127.0.0.2:1": 1,
+                        "127.0.0.1:1": 1
+
+                    },
+                    "retries": 2,
+                    "type": "roundrobin"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 13: hit routes, retries failed, status code is 502
+--- request
+GET /hello
+--- error_code: 502
+--- grep_error_log eval
+qr/X-Apisix-Upstream-Status: 502/
+--- grep_error_log_out
+X-Apisix-Upstream-Status: 502
+
+
+
+=== TEST 14: the status code returned from apisix
+--- config
+    location /t {
+        content_by_lua_block {
+            ngx.exit(500)
+        }
+    }
+--- request
+GET /t
+--- error_code: 500
+--- grep_error_log eval
+qr/X-Apisix-Upstream-Status:/
+--- grep_error_log_out
+
+
+
+=== TEST 15: the status code returned from apisix
+--- config
+    location /t {
```

Comment:
This route doesn't belong to APISIX. You can confirm it via looking at the generated `t/servroot/conf/nginx.conf`.
You can use `limit-count` plugin to generate a 5xx from APISIX.

Reply:
Nice, I will update later.